perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/FacilityAttributeProcessor.java

```diff
@@ -86,4 +91,24 @@ public void processAllAttributesRemoved(String msg, MessageBeans beans) {
 		}
 	}
 
-}
+	public void processVirtualAttributeChanged(String msg, MessageBeans beans) {
+		if (beans.getUser() == null || beans.getAttribute() == null || beans.getResource() == null) {
+			return;
+		}
+		try {
+			log.debug("Getting resources assigned to facility {}", beans.getFacility().getId());
+			List<Resource> resources = ldapcManager.getPerunBl().getFacilitiesManager().getAssignedResources(ldapcManager.getPerunSession(), beans.getFacility());
```

Comment:
use ``getFacilitiesManagerBl()`` you will get rid of ``PrivilegeException``

Reply:
The problem is that object in LdapcManager is the type Perun and not PerunBl so I can't use getFacilitiesManagerBl(). Should I rename methods to from get/setPerunBl to get/setPerun or just change type to PerunBl?